core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -1235,14 +1238,14 @@ private StoreConfiguration getStoreConfig(Object store) {
    class StoreStatus {
       final Object store;
       final StoreConfiguration config;
-      volatile boolean availability = true;
+      boolean availability = true;
 
       StoreStatus(Object store, StoreConfiguration config) {
          this.store = store;
          this.config = config;
       }
 
-      boolean availabilityChanged() {
+      synchronized boolean availabilityChanged() {
```

Comment:
The locking here is a bit questionable, but the subsequent read is on the same thread so it is fine.

Reply:
I also don't like the fact that the name of the method is worded like it's a simple getter, but it can block for a long time (like it happened with the `RocksDB` store: https://issues.jboss.org/browse/ISPN-9762)

I would have preferred the lists of loaders/writers to hold a wrapper that contained the loader/writer reference + status + configuration, but it would only have a `compareAndSet` method, with the actually availability check being triggered directly in the `pollStoreAvailability` method. I think I've already mentioned that I like to avoid map lookups in other discussions :)